src/common/cryptography.js

```diff
@@ -48,7 +48,7 @@ module.exports.hashArray = (inputsArray) => {
   for (let e of inputsArray) {
     e = utils.remove0x(e);
     // since the inputs are in hex string, they are twice as long as their bytes
-    hexStr += (new BN(e.length/2).toBuffer('be', 8).toString('hex')) + e;
+    hexStr += JSBI.BigInt(e.length/2).toString(16).padStart(16, '0') + e;
```

Comment:
I really preferred that it was explicitly saying BigEndian, but maybe this can be addressed in a different PR

Reply:
@lenak25

Reply:
I understand, and I agree with you @elichai. I looked into JSBI, and there seems to be no explicit mention to "endianness" anywhere in the repo, so it must be using big endian by default, and there seems to be no way of specifying it.